UITextView+Placeholder/UITextView+Placeholder.m

```diff
@@ -55,14 +55,7 @@ - (void)swizzledDealloc {
 #pragma mark `defaultPlaceholderColor`
 
 + (UIColor *)defaultPlaceholderColor {
-    static UIColor *color = nil;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        UITextField *textField = [[UITextField alloc] init];
-        textField.placeholder = @" ";
-        color = [textField valueForKeyPath:@"_placeholderLabel.textColor"];
-    });
-    return color;
+    return [UIColor colorWithWhite:0.70f alpha:1.0f];
```

Comment:
I'm not sure the follow you. The default color for placeholder (as stated [here](https://developer.apple.com/library/prerelease/ios/documentation/UIKit/Reference/UITextField_Class/index.html#//apple_ref/occ/instp/UITextField/placeholder)) is 70% grey.

I do agree that your method is more practical, as Apple may change the default color of placeholder at any time, but your are using the private API to obtain the color which is not allowed.